codice/ddf
# DDF-04507 Refactored filter-builder to use backbone model / collection

#### What does this PR do?
This is the first part of a refactor with filter-builder to rip out functions from the views and migrate them to models and collections. This will make the migration to React much easier when listening to changes on the model.

This greatly cleans up the flow of how filters get rehydrated within the view and how the data is serialized/deserialized within Intrigue.
#### Who is reviewing it? 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@djblue 
@andrewkfiedler 
@middlets719 
@gjvera 
@zta6 

#### How should this be tested?

Create queries from advanced builder and ensure they perform as normal.

- Nested groupings, change the operator, add filters, save them and make sure they reload.

#### Any background context you want to provide?
Overall pre-improvement before migrating to react.

#### What are the relevant tickets?
For GH Issues:
Fixes: #4507
